rnr_node/src/rnr_node.c

```diff
@@ -0,0 +1,960 @@
+/*
+ * Copyright (c) 2016 PolySync
+ *
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+
+/**
+ * \example rnr_node.c
+ *
+ * Record and Replay Participant Node Example.
+ *
+ * Shows how to use RnR and Logfile APIs to handle record/replay operations.
+ *
+ * It will log example ps_event_msg's when recording.
+ *
+ * The example uses the standard PolySync node template and state machine.
+ * Send the SIGINT (control-C on the keyboard) signal to the node/process to do a graceful shutdown.
+ * See \ref polysync_node_template.h for more information.
+ *
+ */
+
+
+
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <signal.h>
+#include <unistd.h>
+#include <glib-2.0/glib.h>
+
+// API headers
+#include "polysync_core.h"
+#include "polysync_node.h"
+#include "polysync_message.h"
+#include "polysync_sdf.h"
+#include "polysync_logfile.h"
+#include "polysync_rnr.h"
+#include "polysync_node_template.h"
+
+
+
+
+// *****************************************************
+// static global types/macros
+// *****************************************************
+
+
+/**
+ * @brief PolySync node name.
+ *
+ */
+static const char NODE_NAME[] = "polysync-rnr-node-c";
+
+
+/**
+ * @brief How often to log message data. [Hertz]
+ *
+ */
+static const ps_timestamp WRITE_FREQ = 10;
+
+
+/**
+ * @brief TODO message type name.
+ *
+ */
+static const char RNR_MSG_NAME[] = "ps_rnr_msg";
+
+#warning "TODO - docs on these constants and docs on declarations/updates"
```

Comment:
Is this still to be done? If so, this should probably be tracked elsewhere so that the examples don't compile with warnings.